app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.java

```diff
@@ -976,33 +992,48 @@ private void sendFetchTimelineRequest(@Nullable String maxId, @Nullable String s
             } else {
                 mode = TimelineRequestMode.NETWORK;
             }
-            timelineRepo.getStatuses(maxId, sinceId, sinceIdMinusOne, LOAD_AT_ONCE, mode)
-                    .observeOn(AndroidSchedulers.mainThread())
+
+            Single<?> result =
+                    timelineRepo.getStatuses(maxId, sinceId, sinceIdMinusOne, LOAD_AT_ONCE, mode)
+                            .observeOn(AndroidSchedulers.mainThread())
+                            .doOnSuccess((response) -> onFetchTimelineSuccess(response, fetchEnd, pos))
+                            .doOnError((err) -> onFetchTimelineFailure(new Exception(err), fetchEnd, pos));
+
+            result
                     .as(autoDisposable(from(this, Lifecycle.Event.ON_DESTROY)))
-                    .subscribe(
-                            (result) -> onFetchTimelineSuccess(result, fetchEnd, pos),
-                            (err) -> onFetchTimelineFailure(new Exception(err), fetchEnd, pos)
-                    );
+                    .subscribe();
+            return result;
         } else {
-            Callback<List<Status>> callback = new Callback<List<Status>>() {
-                @Override
-                public void onResponse(@NonNull Call<List<Status>> call, @NonNull Response<List<Status>> response) {
-                    if (response.isSuccessful()) {
-                        onFetchTimelineSuccess(liftStatusList(response.body()), fetchEnd, pos);
-                    } else {
-                        onFetchTimelineFailure(new Exception(response.message()), fetchEnd, pos);
-                    }
-                }
+            Single<?> result = Single
```

Comment:
Maybe I shouldn't do this and should instead convert all of the timeline requests into `Single`s.